test/complex.jl

```diff
@@ -800,6 +800,7 @@ end
 
 @test complex.(1.0, [1.0, 1.0]) == [complex(1.0, 1.0), complex(1.0, 1.0)]
 @test complex.([1.0, 1.0], 1.0) == [complex(1.0, 1.0), complex(1.0, 1.0)]
+@test_throws DimensionMismatch complex(zeros(Int, 2), zeros(Float64, 3))
```

Comment:
Should also add tests for the new functionality, and maybe add a test for array-of-arrays.